draft-barnes-acme.md

```diff
@@ -536,121 +508,101 @@ Recovery tokens are employed in response to Recovery Challenges.  Such challenge
 
-type (required, string):
-: "certificate"
-
-certificate (required, string):
-: The issued certificate, as a base64-encoded DER certificate.
-
-chain (optional, array of string):
-: A chain of CA certificates which are parents of the issued certificate.  Each certificate is in base64-encoded DER form (not PEM, as for CSRs above).  This array MUST be presented in the same order as would be required in a TLS handshake {{RFC5246}}.
+The default format of the certificate is DER (application/pkix-cert).  The client may request other formats by including an Accept header in its request.
 
-refresh (optional, string):
-: An HTTP or HTTPS URI from which updated versions of this certificate can be fetched.
+The server can provide metadata about the certificate in HTTP headers.  For example, the server can include a Link header {{RFC5988}} with relation "up" to provide a certificate under which this certificate was issued.  Or the server can include an Expires header as a hint to the client about when to re-query to refresh the certificate.  (Of course, the real expiration of the certificate is controlled by the notAfter time in the certificate itself.)
 
 ~~~~~~~~~~
 
-{
-  "type": "certificate",
-  "certificate": "Zmzdx7UKvwDJ6bk...YBX22NPGQZyYcg",
-  "chain": [
-    "WUn8L2vLT553pIWJ2...gJ574o2anls1k2p",
-    "y3O4puZa9r5KBk1LX...Ya7jlaAZUfuYZGZ"
-  ],
-  "refresh": "https://example.com/refresh/Dr8eAwTVQfSS/"
-}
+HTTP/1.1 201 Created
+Content-Type: application/pkix-cert
+Link: <https://example.com/acme/ca-cert>;rel="up"
+Location: https://example.com/acme/cert/asdf
+
+[DER-encoded certificate]
 
 ~~~~~~~~~~
 
-The certificate message allows the server to provide the certificate itself, as well as some associated management information.  The chain of CA certificates can simplify TLS server configuration, by allowing the server to suggest the certificate chain that a TLS server using the issued certificate should present.
+## Certificate Refresh
 
-The refresh URI allows the client to download updated versions of the issued certificate, in the sense of certificates with different validity intervals, but otherwise the same contents (in particular, the same names and public key).  This can be useful in cases where a CA wishes to issue short-lived certificates, but is still willing to vouch for an identifier-key binding over a longer period of time.  To download an updated certificate, the client simply sends a GET request to the refresh URI.
+The certificate URL (provided in the Location header of the server's response) is used to refresh or revoke the certificate.  To refresh the certificate, the client simply sends a GET request to the certificate URL.  This allows the server to provide the client with updated certificates with the same content and different validity intervals, for as long as the server considers the client authorized.
```

Comment:
Agreed. And this may actually refer to a different authorization period than the one we discuss before. For instance, if you do a DV identity validation on startssl.com, that validation lasts 30 days, and you must request a certificate within those 30 days. But that certificate then lasts a year. Are we going to have a similar notion of different validity periods? I think so, in which case we should define them.